aodndata/aatams_sattag_qc_dm/aatams_sattag_qc_dm_schema.py

```diff
@@ -0,0 +1,685 @@
+import os
+from datetime import datetime
+from functools import partial
+
+from schema import And, Or, Schema, Use
+
+from .csv_schema import CSVSchema
+
+AATAMS_DM_NUMBER_OF_FILES_IN_ZIP = 7
+AATAMS_DM_FILE_TYPE_NAMES = (
+    "metadata",
+    "ctd",
+    "diag",
+    "dive",
+    "haulout",
+    "ssmoutputs",
+    "summary",
+)
+
+DATE_FMT_STR_1 = "%Y-%m-%dT%H:%M:%SZ"
+DATE_FMT_STR_2 = "%m/%d/%y %H:%M:%S"
+
+
+# schema functions -> Return arg or False
+def valid_device(astr):  # TODO expand to particular devices fmts?
+    """Check if a valid device.
+
+    Args:
+      astr: a device string
+
+    Returns:
+      str: astr if valid
+      bool: False if invalid
+
+    """
+    return astr if len(astr.split("-")) >= 3 else False
+
+
+def valid_wmo_device(astr):  # TODO expand to particular codes?
+    """Check if a valid wmo device.
+
+    Args:
+      astr: a wmo device string
+
+    Returns:
+      str: astr if valid
+      bool: False if invalid
+
+    """
+    return astr if astr[0] == "Q" else False
+
+
+def valid_tag_type(astr):  # TODO expand to particular strings.
+    """Check if a valid tag type.
+
+    Args:
+      astr: a tag type string
+
+    Returns:
+      str: astr if valid
+      bool: False if invalid
+
+    """
+    return astr if "_" in astr else False
```

Comment:
TODO expand to particular strings. -- I would remove this part, with the doco I have I can't guarantee they all have a '_'

Reply:
ok